### PR TITLE
G-g-g-g-g-g-g-g-ghooooosts!

### LINKED ISSUE #680: G-G-G-G-Ghooooosts!
Basically, dead players should be able to become observers. Also, deadchat, ghost sprites that can only be seen by other ghosts, etc.

### PR DESCRIPTION
**Pull request by Zumorica**
Closes #680 
Pretty basic ghosts. Includes deadchat, and sprites that can only be seen by ghosts.
~~Requires: https://github.com/space-wizards/RobustToolbox/pull/1019~~

### CODE REVIEW

**Comment by Zumorica** [comment]
This is ready for review now.

File: Content.Client/GameObjects/Components/Observer/GhostComponent.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            switch (message)
+            {
+                case PlayerAttachedMsg _:
+                    if (_gui == null)
+                    {
+                        _gui = new GhostGui(this);
+                    }
+                    else
+                    {
+                        _gui.Parent?.RemoveChild(_gui);
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Observer/GhostComponent.cs:72]:
```suggestion
                        _gui.Orphan();
```

File: Content.Server/Chat/ChatManager.cs [hunk 2]
```diff
@@ -93,6 +95,17 @@ public void SendOOC(IPlayerSession player, string message)
...
         }
 
+        public void SendDeadChat(IPlayerSession player, string message)
+        {
+            var clients = _playerManager.GetPlayersBy(x => x.AttachedEntity != null && x.AttachedEntity.HasComponent<GhostComponent>()).Select(p => p.ConnectedClient);;
+
+            var msg = _netManager.CreateNetMessage<MsgChatMessage>();
+            msg.Channel = ChatChannel.Dead;
+            msg.Message = message;
+            msg.MessageWrap = $"DEAD: {player.AttachedEntity.Name}: {{0}}";
```
Comment (PJB3005) [inline Content.Server/Chat/ChatManager.cs:105]: This should be localized.

File: Content.Server/Chat/ChatManager.cs [hunk 3]
```diff
@@ -93,6 +95,17 @@ public void SendOOC(IPlayerSession player, string message)
             _mommiLink.SendOOCMessage(player.SessionId.ToString(), message);
         }
 
+        public void SendDeadChat(IPlayerSession player, string message)
+        {
+            var clients = _playerManager.GetPlayersBy(x => x.AttachedEntity != null && x.AttachedEntity.HasComponent<GhostComponent>()).Select(p => p.ConnectedClient);;
+
+            var msg = _netManager.CreateNetMessage<MsgChatMessage>();
```
Comment (PJB3005) [inline Content.Server/Chat/ChatManager.cs:102]: No speech bubbles?

File: Content.Client/GameObjects/Components/Observer/GhostComponent.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    public class GhostComponent : SharedGhostComponent
+    {
+        private GhostGui _gui;
+        private bool _canReturnToBody = true;
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public override bool CanReturnToBody
+        {
+            get => _canReturnToBody;
+            set {}
```
[the remark this answers is not in the document]
Comment (Zumorica) [reply Content.Client/GameObjects/Components/Observer/GhostComponent.cs]: Yeah, you're right. Removed it from shared

File: Content.Client/GameObjects/Components/Observer/GhostComponent.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        {
+            base.OnRemove();
+
+            _gui?.Dispose();
+        }
+
+
+        private void SetGhostVisibility(bool visibility)
+        {
+            // So, for now this is a client-side hack... Please, PLEASE someone make this work server-side.
```
[the remark this answers is not in the document]
Comment (Zumorica) [reply Content.Client/GameObjects/Components/Observer/GhostComponent.cs:37]: Yeah. Maybe we could implement like "masks and layers" specifically for PVS?
But anyway, we might want to open an issue for this.

File: Content.Server/Chat/ChatManager.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/Chat/ChatManager.cs]: Done.

File: Content.Server/Chat/ChatManager.cs (hunk 3 above)
Comment (Zumorica) [reply Content.Server/Chat/ChatManager.cs:106]: Whoops, I completely forgot we had fancy speech bubbles when coding this--
Added them now.

**Pull request merged**